#VRML_SIM R2020a utf8
WorldInfo {
  contactProperties [
    ContactProperties {
      material1 "track"
      material2 "floor"
      coulombFriction [
        500
      ]
      softCFM 1e-05
    }
  ]
}
Viewpoint {
  orientation -0.5896652172158823 -0.5692675973152281 -0.5729130250332014 2.12411324874228
  position 0.19182441958846388 3.0972216152165717 0.015957426036962825
}
TexturedBackground {
  texture "noon_park_empty"
}
TexturedBackgroundLight {
  texture "noon_park_empty"
}
DEF Floor Solid {
  translation 0 -0.5 0
  children [
    DEF FLOOR_BOUNDARY Shape {
      appearance PBRAppearance {
        baseColor 0.2 0.2 0.2
        roughness 1
        metalness 0
      }
      geometry Box {
        size 10 1 10
      }
    }
  ]
  name "floor"
  contactMaterial "floor"
  boundingObject USE FLOOR_BOUNDARY
}
StraightRoadSegment {
  translation 0 0.0001 -2
  width 0.1
  lines [
    RoadLine {
      type "none"
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  length 5
  appearance NULL
}
StraightRoadSegment {
  translation -3 0.0001 0
  rotation 0 1 0 1.5708003061004252
  name "road(1)"
  width 0.1
  lines [
    RoadLine {
      type "none"
    }
  ]
  rightBorder FALSE
  leftBorder FALSE
  length 5
  appearance NULL
}
Robot {
  translation 0 0.03 0
  children [
    HingeJoint {
      jointParameters HingeJointParameters {
        axis 0 1 0
        anchor 0 0.07 0
      }
      device [
        RotationalMotor {
          name "tower rotational motor"
        }
        PositionSensor {
          name "tower angle sensor"
        }
      ]
      endPoint Camera {
        translation 0 0.15 -6.0292e-14
        rotation -0.12942694302206584 -0.9831055672050736 -0.12942994302074512 1.58784
        width 160
        height 160
      }
    }
    Compass {
      yAxis FALSE
    }
    DistanceSensor {
      translation 0.075 0 0
    }
    Track {
      translation 0.08 0 -0.11
      children [
        DEF WHEEL1 TrackWheel {
          radius 0.025
          children [
            DEF WHEEL1_TRANSFORM Transform {
              rotation 1 0 0 1.5707996938995747
              children [
                DEF WHEEL1_BOUNDARY Shape {
                  appearance DEF WHEEL_COLOUR PBRAppearance {
                    roughness 1
                    metalness 0.5
                  }
                  geometry Cylinder {
                    height 0.045
                    radius 0.025
                  }
                }
              ]
            }
          ]
        }
        DEF WHEEL2 TrackWheel {
          position -0.15 0
          radius 0.025
          children [
            USE WHEEL1_TRANSFORM
          ]
        }
        DEF WHEEL3 TrackWheel {
          position -0.085 0.049
          radius 0.02
          children [
            Transform {
              rotation 1 0 0 1.5707996938995747
              children [
                DEF WHEEL3_BOUNDARY Shape {
                  appearance USE WHEEL_COLOUR
                  geometry Cylinder {
                    height 0.045
                    radius 0.02
                  }
                }
              ]
            }
          ]
        }
      ]
      name "left track"
      contactMaterial "track"
      boundingObject DEF TRACK_BOUNDARY Group {
        children [
          Transform {
            rotation 1 0 0 1.5707996938995747
            children [
              USE WHEEL1_BOUNDARY
            ]
          }
          Transform {
            translation -0.15 0 0
            rotation 1 0 0 1.5707996938995747
            children [
              USE WHEEL1_BOUNDARY
            ]
          }
          Transform {
            translation -0.085 0.049 0
            rotation 1 0 0 1.5707996938995747
            children [
              USE WHEEL3_BOUNDARY
            ]
          }
          Transform {
            translation -0.075 -0.0125 0
            children [
              Shape {
                geometry Box {
                  size 0.15 0.025 0.045
                }
              }
            ]
          }
        ]
      }
      physics Physics {
      }
      device [
        LinearMotor {
          name "left motor"
        }
      ]
      animatedGeometry DEF ANIMATION_GROUP Group {
        children [
          Shape {
            appearance DEF TRACK_ANI_COLOUR PBRAppearance {
              baseColor 0.2 0.2 0.2
              roughness 0.3
            }
            geometry Box {
              size 0.01 0.003 0.045
            }
          }
          Transform {
            translation 0 0.0023 0
            children [
              Shape {
                appearance USE TRACK_ANI_COLOUR
                geometry Box {
                  size 0.003 0.0016 0.045
                }
              }
            ]
          }
          Transform {
            translation 0.007 0 0.02
            children [
              DEF TRACK_SIDE_ANI_SHAPE Shape {
                appearance USE TRACK_ANI_COLOUR
                geometry Box {
                  size 0.004 0.003 0.003
                }
              }
            ]
          }
          Transform {
            translation 0.007 0 -0.02
            children [
              USE TRACK_SIDE_ANI_SHAPE
            ]
          }
        ]
      }
      geometriesCount 35
    }
    Track {
      translation 0.08 0 0.11
      children [
        USE WHEEL1
        USE WHEEL2
        USE WHEEL3
      ]
      name "right track"
      contactMaterial "track"
      boundingObject USE TRACK_BOUNDARY
      physics Physics {
      }
      device [
        LinearMotor {
          name "right motor"
        }
      ]
      animatedGeometry USE ANIMATION_GROUP
      geometriesCount 35
    }
  ]
  boundingObject Group {
    children [
      Transform {
        translation 0 0.03 0
        children [
          Shape {
            geometry Box {
              size 0.15 0.07 0.17
            }
          }
        ]
      }
    ]
  }
  physics Physics {
  }
  controller "<extern>"
}
